pkg/executor/plugins/grpc/proto/executor.proto

```diff
@@ -0,0 +1,125 @@
+
+syntax = "proto3";
+package proto;
+
+message RunCommandRequest {
+  bytes Params = 1;
+}
+
+message RunCommandResponse {
+  bytes Params = 1;
+}
+
+message CancelCommandRequest {
+  string ExecutionID =1;
+}
+
+message CancelCommandResponse {
+
+}
+
+message IsInstalledRequest {
+
+}
+
+message IsInstalledResponse {
+  bool Installed = 1;
+}
+
+message ShouldBidRequest {
+  bytes BidRequest = 1;
+}
+
+message ShouldBidBasedOnUsageRequest {
+  bytes BidRequest = 1;
+  bytes Usage = 2;
```

Comment:
Similar to a comment I left above, we should define the actual `BidRequest` and `Usage` fields in the proto schema instead of just bytes.

Also a similar comment I left for `JobStore` PR that @rossjones is working on. We should wrap requests (and possibly responses) in an envelope to help maintain backward compatibility with future changes. Meaning all requests that we send over the write should be wrapped in an envelope that might look like:

```
syntax = "proto3";

message Envelope {
    string version = 1;
    string timestamp = 2;
    Metadata metadata = 3;
    bytes payload = 4; // or maybe use `any`
    string checksum = 5;

    message Metadata {
        string serialization_format = 1;
        // ... any other metadata fields you need.
    }
}

```

**Envelope Schema**

- version: (String) Represents the version of the schema used for the payload. When a backward-incompatible change to the object schema occurs, this version number can be incremented. Code that reads this data can then decide on different processing paths based on the version.

- timestamp: (String) Represents the time at which the data was stored or transmitted. It's typically in ISO 8601 format, for example "2023-07-30T13:45:00Z". This can aid in conflict resolution, debugging, and tracing.

- metadata: (Object) This is a map that stores additional contextual information about the object or the transmission. It can include fields such as:
  - serialization_format: (String) This specifies the serialization format used for the payload. It allows for the payload to be properly deserialized, even if the serialization method changes in the future.
  - Additional fields can be added as required to provide further contextual information about the payload or the transmission.
- payload: (Byte Array) This contains the serialized form of the actual object being stored or transmitted. The method of serialization is indicated in the serialization_format field in the metadata object.

- checksum: (String) This contains a cryptographic hash of the payload. It's used to verify the integrity of the payload data, ensuring it hasn't been tampered with during storage or transmission. When the data is read, the hash of the payload can be computed and compared to this value to verify they match.

This schema is designed to be flexible and can be extended or adjusted based on specific needs and constraints. The primary goal is to ensure backward compatibility, maintain data integrity, and provide flexibility for future changes in serialization methods and object schema.

Reply:
Adding an Envelope sounds good, I can make that change in this PR. However, as I mentioned [above](https://github.com/bacalhau-project/bacalhau/pull/2637#discussion_r1279665465) I am hesitant to implement protobuf messages for all of our models at this point in implementation.

Would serializing the go structure as json bytes in the envelop payload field be a reasonable compromise?

Reply:
Filed issue for this work in https://github.com/bacalhau-project/bacalhau/issues/2701